qiskit/circuit/quantumcircuit.py

```diff
@@ -2636,14 +2636,26 @@ def parameters(self) -> ParameterView:
     @property
     def num_parameters(self) -> int:
         """The number of parameter objects in the circuit."""
+        # Avoid a (potential) object creation if we can.
+        if self._parameters is not None:
+            return len(self._parameters)
         return len(self._unsorted_parameters())
 
     def _unsorted_parameters(self) -> set[Parameter]:
-        """Efficiently get all parameters in the circuit, without any sorting overhead."""
-        parameters = set(self._parameter_table)
-        if isinstance(self.global_phase, ParameterExpression):
-            parameters.update(self.global_phase.parameters)
+        """Efficiently get all parameters in the circuit, without any sorting overhead.
+
+        .. warning::
 
+            The returned object may directly view onto the ``ParameterTable`` internals, and so
+            should not be mutated.  This is an internal performance detail.  Code outside of this
+            package should not use this method.
```

Comment:
Do you want to link the object here?
```suggestion
            The returned object may directly view onto the :class:`.ParameterTable` internals, and so
            should not be mutated.  This is an internal performance detail.  Code outside of this
            package should not use this method.
```

Reply:
`ParameterTable` isn't part of the documented API, since it's only meant to be used in the private `QuantumCircuit._parameters`, so there's no actual cross-ref that'll work for it.  That said, this whole docstring is for a private method and not documented itself haha.